scripts/tests/yaml/runner.py

```diff
@@ -0,0 +1,294 @@
+    f = click.option("--test_directory", type=click.Path(exists=True), required=True, show_default=True,
+                     default=TestsFinder.get_default_directory(), help='Path to the directory containing tests.')(f)
+    f = click.option("--specifications_paths", type=click.Path(), required=False, show_default=True,
+                     default="src/app/zap-templates/zcl/data-model/chip/*.xml", help='Path to a set of files containing clusters definitions.')(f)
+    f = click.option("--PICS", type=click.Path(exists=True), required=False, show_default=True,
+                     default='src/app/tests/suites/certification/ci-pics-values', help='Path to the PICS file to use.')(f)
+    f = click.option("--stop_on_error", type=bool, required=False, show_default=True,
+                     default=True, help='Stop parsing on first error.')(f)
+    return f
+
+
+def test_runner_options(f):
+    f = click.option("--adapter", type=str, default=None, required='True',
+                     show_default=True, help='The adapter to run the test with.')(f)
+    f = click.option("--stop_on_error", type=bool, default=True, show_default=True,
+                     help='Stop the test suite on first error.')(f)
+    f = click.option("--stop_on_warning", type=bool, default=False, show_default=True,
+                     help='Stop the test suite on first warning.')(f)
+    f = click.option("--stop_at_number", type=int, default=-1, show_default=True,
+                     help='Stop the the test suite at the specified test number.')(f)
+    f = click.option("--show_adapter_logs", type=bool, default=False, show_default=True,
+                     help='Show additional logs provided by the adapter.')(f)
+    f = click.option("--show_adapter_logs_on_error", type=bool, default=True, show_default=True,
+                     help='Show additional logs provided by the adapter on error.')(f)
+    return f
+
+
+def websocket_runner_options(f):
+    f = click.option("--server_address", type=str, default="localhost", show_default=True,
+                     help='The websocket server address to connect to.')(f)
+    f = click.option("--server_port", type=int, default=9002, show_default=True,
+                     help='The websocket server port to connect to.')(f)
+    f = click.option("--server_name", type=str, default=None,
+                     required=False, help='Name of a websocket server to run at launch.')(f)
+    f = click.option("--server_path", type=click.Path(exists=True), default=None,
+                     required=False, help='Path to a websocket server to run at launch.')(f)
+    f = click.option("--server_arguments", type=str, default=None, required=False,
+                     help='Optional arguments to pass to the websocket server at launch.')(f)
+    return f
+
+
+pass_parser_builder_config = click.make_pass_decorator(TestParserBuilderConfig)
+
+
+class YamlTestParserGroup(click.Group):
```

Comment:
A doc string with a brief explanation of why this is needed would really help save at least 5 mins to someone reviewing this for the first time.